src/test/java/gumtree/spoon/AstComparatorTest.java

```diff
@@ -1771,7 +1771,7 @@ public void testExtends() throws Exception {
 		result.debugInformation();
 
 		assertEquals(1, actions.size());
-		assertTrue(result.containsOperation(OperationKind.Update, "SUPER_TYPE", "SuperClass1"));
```

Comment:
why do we change the role name? AFAIU, it should stay SUPER_TYPE.

Reply:
I am now letting the `TreeScanner` add this node in the tree and the scanner adds the node using `nodeTypeName` which is `TypeReference` for `SuperClass2`. 

I am planning to split this PR into multiple smaller ones because apparently a lot of things have been affected. The `CtTypeReference` in `String s` is confused with `SuperClass2` (super classes) so I think I will have to rethink how I parsed nested type references.